### DISCUSSION TITLE
passing handelbar list

### DISCUSSION THREAD

**Discussion by sergedahdouh** [handlebars-lang/handlebars.js#1889] [Q&A]
hello i have the following template 

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    
    
    <title>Invoice - Bootdey.com</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
	<script src="https://code.jquery.com/jquery-1.10.2.min.js"></script>
    <link href="https://netdna.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet">
	<script src="https://netdna.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
</head>
<body>
<div class="col-md-12">   
 <div class="row">
		
        <div class="receipt-main col-xs-10 col-sm-10 col-md-6 col-xs-offset-1 col-sm-offset-1 col-md-offset-3">
            <div class="row">
    			<div class="receipt-header">
					<div class="col-xs-6 col-sm-6 col-md-6">
						<div class="receipt-left">
							<img class="img-responsive" alt="iamgurdeeposahan" src="https://bootdey.com/img/Content/avatar/avatar6.png" style="width: 71px; border-radius: 43px;">
						</div>
					</div>
					<div class="col-xs-6 col-sm-6 col-md-6 text-right">
						<div class="receipt-right">
							<h5>{{shopName}}</h5>
							<p>{{shopPhone}} <i class="fa fa-phone"></i></p>
							<p>{{shopEmail}} <i class="fa fa-envelope-o"></i></p>
							<p>USA <i class="fa fa-location-arrow"></i></p>
						</div>
					</div>
				</div>
            </div>
			
			<div class="row">
				<div class="receipt-header receipt-header-mid">
					<div class="col-xs-8 col-sm-8 col-md-8 text-left">
						<div class="receipt-right">
							<h5>Customer Name </h5>
							<p><b>Mobile :</b> +1 12345-4569</p>
							<p><b>Email :</b> customer@gmail.com</p>
							<p><b>Address :</b> New York, USA</p>
						</div>
					</div>
					<div class="col-xs-4 col-sm-4 col-md-4">
						<div class="receipt-left">
							<h3>INVOICE # 102</h3>
						</div>
					</div>
				</div>
            </div>
			
            <div>
                <table class="table table-bordered">
                    <thead>
                        <tr>
                            <th>Description</th>
                            <th>Amount</th>
                        </tr>
                    </thead>
                    <tbody>
						{{#items}}

                        <tr>
                            <td class="col-md-9">{{User_Name}}</td>
                            <td class="col-md-3"><i class="fa fa-inr"></i> {{itemPrice}}</td>
                        </tr>
                       
                        {{/items}}

                        <tr>
                            <td class="text-right">
                            <p>
                                <strong>Total Amount: </strong>
                            </p>
                            <p>
                                <strong>Late Fees: </strong>
                            </p>
							<p>
                                <strong>Payable Amount: </strong>
                            </p>
							<p>
                                <strong>Balance Due: </strong>
                            </p>
							</td>
                            <td>
                            <p>
                                <strong><i class="fa fa-inr"></i> 65,500/-</strong>
                            </p>
                            <p>
                                <strong><i class="fa fa-inr"></i> 500/-</strong>
                            </p>
							<p>
                                <strong><i class="fa fa-inr"></i> 1300/-</strong>
                            </p>
							<p>
                                <strong><i class="fa fa-inr"></i> 9500/-</strong>
                            </p>
							</td>
                        </tr>
                        <tr>
                           
                            <td class="text-right"><h2><strong>Total: </strong></h2></td>
                            <td class="text-left text-danger"><h2><strong><i class="fa fa-inr"></i> 31.566/-</strong></h2></td>
                        </tr>
                    </tbody>
                </table>
            </div>
			
			<div class="row">
				<div class="receipt-header receipt-header-mid receipt-footer">
					<div class="col-xs-8 col-sm-8 col-md-8 text-left">
						<div class="receipt-right">
							<p><b>Date :</b> 15 Aug 2016</p>
							<h5 style="color: rgb(140, 140, 140);">Thanks for shopping.!</h5>
						</div>
					</div>
					<div class="col-xs-4 col-sm-4 col-md-4">
						<div class="receipt-left">
							<h1>Stamp</h1>
						</div>
					</div>
				</div>
            </div>
			
        </div>    
	</div>
</div>

<style type="text/css">
body{
background:#eee;
margin-top:20px;
}
.text-danger strong {
        	color: #9f181c;
		}
		.receipt-main {
			background: #ffffff none repeat scroll 0 0;
			border-bottom: 12px solid #333333;
			border-top: 12px solid #9f181c;
			margin-top: 50px;
			margin-bottom: 50px;
			padding: 40px 30px !important;
			position: relative;
			box-shadow: 0 1px 21px #acacac;
			color: #333333;
			font-family: open sans;
		}
		.receipt-main p {
			color: #333333;
			font-family: open sans;
			line-height: 1.42857;
		}
		.receipt-footer h1 {
			font-size: 15px;
			font-weight: 400 !important;
			margin: 0 !important;
		}
		.receipt-main::after {
			background: #414143 none repeat scroll 0 0;
			content: "";
			height: 5px;
			left: 0;
			position: absolute;
			right: 0;
			top: -13px;
		}
		.receipt-main thead {
			background: #414143 none repeat scroll 0 0;
		}
		.receipt-main thead th {
			color:#fff;
		}
		.receipt-right h5 {
			font-size: 16px;
			font-weight: bold;
			margin: 0 0 7px 0;
		}
		.receipt-right p {
			font-size: 12px;
			margin: 0px;
		}
		.receipt-right p i {
			text-align: center;
			width: 18px;
		}
		.receipt-main td {
			padding: 9px 20px !important;
		}
		.receipt-main th {
			padding: 13px 20px !important;
		}
		.receipt-main td {
			font-size: 13px;
			font-weight: initial !important;
		}
		.receipt-main td p:last-child {
			margin: 0;
			padding: 0;
		}	
		.receipt-main td h2 {
			font-size: 20px;
			font-weight: 900;
			margin: 0;
			text-transform: uppercase;
		}
		.receipt-header-mid .receipt-left h1 {
			font-weight: 100;
			margin: 34px 0 0;
			text-align: right;
			text-transform: uppercase;
		}
		.receipt-header-mid {
			margin: 24px 0;
			overflow: hidden;
		}
		
		#container {
			background-color: #dcdcdc;
		}
</style>

<script type="text/javascript">

</script>
</body>
</html>

and i am trying to pass through 

https://us-central1-txxxxx.cloudfunctions.net/ext-pdf-generator-zb4r-executePdfGenerator?templatePath=trackingapp-66e74.appspot.com/index.zip&outputFileName=basic-example.pdf&chromiumPdfOptions[printBackground]=true&adjustHeightToFit=true&data[items][User_Name]=
  [
    "User_Name": "T"
  ,
  
    "User_Name": "R"
  ,
  
    "User_Name": "M"
  
]

the items as list but is never get it as list is always get it in one raw
 [
<img width="752" alt="Screenshot 2022-09-27 at 03 14 46" src="https://user-images.githubusercontent.com/40029858/192408669-ebb0004d-a8a8-4b87-91a9-3e08cd90512b.png">
](url)

**Comment by jaylinski** [comment]
I can maybe help you if you tell me what your exact problem is and what you want to achieve.

**Reply by sergedahdouh** [reply]
thank you so much for your answer what actually trying to do is the following i have Generate PDF using Puppeteer and Handlebars.js as extensions in firebase this extention is to create a pdf document using its api and the template that i have attached the template is using handelbars and the api is the following 

https://us-central1-xx.cloudfunctions.net/ext-pdf-generator-zb4r-executePdfGenerator?templatePath=trackingapp-66e74.appspot.com/index.zip&outputFileName=basic-example.pdf&chromiumPdfOptions[printBackground]=true&adjustHeightToFit=true&data[shopName]=test&data[shopPhone]=3333&data[shopEmail]=zz@hotm&data[items][User_Name]=[
{User_Name":"Product LK","itemPrice":"1000CHF"},{"User_Name":"Product L","itemPrice":"150CHF"},{"User_Name":"Product O","itemPrice":"50CHF}]

but as you can see in the attached is not working 
<img width="752" alt="Screenshot 2022-09-27 at 14 19 49" src="https://user-images.githubusercontent.com/40029858/192524789-9b6e11fa-6efd-4c8a-8345-1ad887d0f272.png">

my goal that every User_Name should be on one row

**Reply by jaylinski** [reply]
You have to use the `#each`-helper. See this example:

https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%3Ctable%3E%5Cn%20%20%20%3Cthead%3E%5Cn%20%20%20%20%20%20%3Ctr%3E%5Cn%20%20%20%20%20%20%20%20%20%3Cth%3EDescription%3C%2Fth%3E%5Cn%20%20%20%20%20%20%20%20%20%3Cth%3EAmount%3C%2Fth%3E%5Cn%20%20%20%20%20%20%3C%2Ftr%3E%5Cn%20%20%20%3C%2Fthead%3E%5Cn%20%20%20%3Ctbody%3E%5Cn%20%20%20%20%20%20%7B%7B%23each%20items%7D%7D%5Cn%20%20%20%20%20%20%3Ctr%3E%5Cn%20%20%20%20%20%20%20%20%20%3Ctd%20class%3D%5C%22col-md-9%5C%22%3E%7B%7BUser_Name%7D%7D%3C%2Ftd%3E%5Cn%20%20%20%20%20%20%20%20%20%3Ctd%20class%3D%5C%22col-md-3%5C%22%3E%3Ci%20class%3D%5C%22fa%20fa-inr%5C%22%3E%3C%2Fi%3E%20%7B%7BitemPrice%7D%7D%3C%2Ftd%3E%5Cn%20%20%20%20%20%20%3C%2Ftr%3E%5Cn%20%20%20%20%20%20%7B%7B%2Feach%7D%7D%5Cn%20%20%20%3C%2Ftbody%3E%5Cn%3C%2Ftable%3E%5Cn%22%2C%22partials%22%3A%5B%5D%2C%22input%22%3A%22%7B%5Cn%20%20items%3A%20%5B%5Cn%5Ct%7B%5C%22User_Name%5C%22%3A%5C%22Product%20LK%5C%22%2C%5C%22itemPrice%5C%22%3A%5C%221000CHF%5C%22%7D%2C%5Cn%5Ct%7B%5C%22User_Name%5C%22%3A%5C%22Product%20L%5C%22%2C%5C%22itemPrice%5C%22%3A%5C%22150CHF%5C%22%7D%2C%5Cn%5Ct%7B%5C%22User_Name%5C%22%3A%5C%22Product%20O%5C%22%2C%5C%22itemPrice%5C%22%3A%5C%2250CHF%5C%22%7D%5Cn%20%20%5D%5Cn%7D%22%2C%22output%22%3A%22%3Ctable%3E%5Cn%20%20%20%3Cthead%3E%5Cn%20%20%20%20%20%20%3Ctr%3E%5Cn%20%20%20%20%20%20%20%20%20%3Cth%3EDescription%3C%2Fth%3E%5Cn%20%20%20%20%20%20%20%20%20%3Cth%3EAmount%3C%2Fth%3E%5Cn%20%20%20%20%20%20%3C%2Ftr%3E%5Cn%20%20%20%3C%2Fthead%3E%5Cn%20%20%20%3Ctbody%3E%5Cn%20%20%20%20%20%20%3Ctr%3E%5Cn%20%20%20%20%20%20%20%20%20%3Ctd%20class%3D%5C%22col-md-9%5C%22%3EProduct%20LK%3C%2Ftd%3E%5Cn%20%20%20%20%20%20%20%20%20%3Ctd%20class%3D%5C%22col-md-3%5C%22%3E%3Ci%20class%3D%5C%22fa%20fa-inr%5C%22%3E%3C%2Fi%3E%201000CHF%3C%2Ftd%3E%5Cn%20%20%20%20%20%20%3C%2Ftr%3E%5Cn%20%20%20%20%20%20%3Ctr%3E%5Cn%20%20%20%20%20%20%20%20%20%3Ctd%20class%3D%5C%22col-md-9%5C%22%3EProduct%20L%3C%2Ftd%3E%5Cn%20%20%20%20%20%20%20%20%20%3Ctd%20class%3D%5C%22col-md-3%5C%22%3E%3Ci%20class%3D%5C%22fa%20fa-inr%5C%22%3E%3C%2Fi%3E%20150CHF%3C%2Ftd%3E%5Cn%20%20%20%20%20%20%3C%2Ftr%3E%5Cn%20%20%20%20%20%20%3Ctr%3E%5Cn%20%20%20%20%20%20%20%20%20%3Ctd%20class%3D%5C%22col-md-9%5C%22%3EProduct%20O%3C%2Ftd%3E%5Cn%20%20%20%20%20%20%20%20%20%3Ctd%20class%3D%5C%22col-md-3%5C%22%3E%3Ci%20class%3D%5C%22fa%20fa-inr%5C%22%3E%3C%2Fi%3E%2050CHF%3C%2Ftd%3E%5Cn%20%20%20%20%20%20%3C%2Ftr%3E%5Cn%20%20%20%3C%2Ftbody%3E%5Cn%3C%2Ftable%3E%5Cn%22%2C%22preparationScript%22%3A%22%22%2C%22handlebarsVersion%22%3A%224.7.7%22%7D
